spec/haskell/src/SEL4/Object/CNode.lhs

```diff
@@ -338,7 +338,7 @@ Determine whether the CTE is valid, and has children to be deleted. If a child i
 >         nextCTE <- withoutPreemption $ getCTE nextPtr
 >         when (cte `isMDBParentOf` nextCTE) $ do
 >             cteDelete nextPtr True
->             preemptionPoint
```

Comment:
Hmm. I should have asked this earlier, but why is this line removed?

Reply:
I thought I put the line back in (was just and editor accident). Let me double check.

Reply:
(Refine did pass here on the PR)

Reply:
(before the squash -- so maybe I messed up something there somehow)

Reply:
Yes, it seems I did, will raise another PR shortly. The version you saw before did not have the line removed (neither did the one I looked at before squashing), but I must have dropped the fix instead of squashing it. Embarrassing.

Reply:
Ok, I double-checked everything else that changed since I last looked at the full diff, and those are jus the edits Mitch recommended, so this line should be the only thing that I messed up.